src/session/content/chat_action_bar.rs

```diff
@@ -282,11 +287,183 @@ impl ChatActionBar {
+                                        | ChatMemberStatus::Administrator(_)
+                                )
+                            } else {
+                                match status.0 {
+                                    ChatMemberStatus::Restricted(data)
+                                        if !data.permissions.can_send_messages =>
+                                    {
+                                        data.permissions.can_send_messages
+                                    }
+                                    ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                    _ => permissions.0.can_send_messages,
+                                }
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+                ChatType::BasicGroup(data) => {
+                    let status_expression =
+                        gtk::ConstantExpression::new(data).chain_property::<BasicGroup>("status");
+                    gtk::ClosureExpression::new::<bool, _, _>(
+                        &[permissions_expression.clone(), status_expression],
+                        closure!(|_: Chat,
+                                  permissions: BoxedChatPermissions,
+                                  status: BoxedChatMemberStatus| {
+                            match status.0 {
+                                ChatMemberStatus::Restricted(data)
+                                    if !data.permissions.can_send_messages =>
+                                {
+                                    data.permissions.can_send_messages
+                                }
+                                ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                _ => permissions.0.can_send_messages,
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+            };
+
+            let message_bar_binding =
+                visible_expresison.bind(&*imp.scrolled_window, "visible", Some(chat));
+            let send_button_binding =
+                visible_expresison.bind(&*imp.send_message_button, "visible", Some(chat));
+            bindings.push(message_bar_binding);
+            bindings.push(send_button_binding);
+
+            // Handle whether or not restriction label should be shown
+            let visible_binding = gtk::ClosureExpression::new::<bool, _, _>(
+                &[chat_expression],
+                closure!(|_: Chat, chat: Chat| {
+                    if !chat.permissions().0.can_send_messages {
+                        if let ChatType::Supergroup(data) = chat.type_() {
+                            if data.is_channel() {
+                                // TODO: implement unmute/mute button
+                                !data.is_channel()
```

Comment:
Have you not checked this 2 lines before? So I think it's fine to just return false here.